src/types/line.js

```diff
@@ -142,9 +144,46 @@ export default class LineAnnotation extends Element {
         y = scaleValue(yScale, options.yMin, y);
         y2 = scaleValue(yScale, options.yMax, y2);
       }
+
+      // Barchart compensation
+      if (chart.config.type === 'bar') {
+        x -= this._compensateAxisForBarchart(chart, xScale, x, true);
+        x2 -= this._compensateAxisForBarchart(chart, xScale, x2, true);
+
+        y -= this._compensateAxisForBarchart(chart, yScale, y, false);
+        y2 -= this._compensateAxisForBarchart(chart, yScale, y2, false);
+      }
     }
     return limitLineToArea({x, y}, {x: x2, y: y2}, chart.chartArea);
   }
+
+  _compensateAxisForBarchart(chart, scale, axis, shouldBeHorizontal) {
+    const isHorizontal = (chart.config.options.indexAxis !== 'y');
```

Comment:
Should be a `===` comparrison instead of `!==` because now its giving the opposite result